src/Controller/OrganisationSponsor/Create.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controller\OrganisationSponsor;
+
+use App\Dto\OrganisationSponsorRequest;
+use App\Form\OrganisationSponsorType;
+use App\Repository\OrganisationSponsor\OrganisationSponsorRepositoryInterface;
+use App\Repository\Organisation\OrganisationRepositoryInterface;
+use App\Service\Event\EventServiceInterface;
+use App\Exceptions\NoEventSelectedException;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Security;
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\Flash\FlashBagInterface;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Routing\RouterInterface;
+use Twig\Environment as Twig;
+
+final class Create
+{
+    private $renderer;
+    private $formFactory;
+    private $repository;
+    private $router;
+    private $eventService;
+    private $organisationRepository;
+    private $flashBag;
+
+    public function __construct(
+        Twig $renderer,
+        FormFactoryInterface $formFactory,
+        OrganisationSponsorRepositoryInterface $repository,
+        OrganisationRepositoryInterface $organisationRepository,
+        RouterInterface $router,
+        EventServiceInterface $eventService,
+        FlashBagInterface $flashBag
+    ) {
+        $this->renderer = $renderer;
```

Comment:
Too many dependencies... creating a service to abstract some of it may help. (cf object calisthenics).